### ISSUE TITLE
Should we support `ops::Add` for `relative::LockTime`

### ISSUE THREAD

**Issue by tcharding** [rust-bitcoin/rust-bitcoin#3676]
Seems reasonable to add together two relative lock times, should we support this?

(If yes then includes adding `iter::Sum`.)

**Comment by apoelstra** [comment]
What should happen when you try to add out of bounds of the type? What if you try to add a height-based locktime to a time-based one?

**Comment by tcharding** [comment]
Ah maybe a better question was should support `Add` for the `units::relative::Height` and `units::relative::Time` (and `iter::Sum`)?

Answering my own question I rekon we should just not touch the `locktime` modules and leave them as is.

FTR we do support custom arithmetic with `BlockInterval`/`BlockHeight`.

**Comment by apoelstra** [comment]
Heh, I'd be fine adding them for `Time` and `Height`, but TBH I don't think anybody ever directly uses those types.

**Comment by tcharding** [comment]
Yeah there is no way to go from a `LockTime` to a `Height` (or `Time`). That is why I said lets not touch it.
